docs/proposal/2022-06-14-configuration-bindings.md

```diff
@@ -0,0 +1,647 @@
+        kubectl:
+          # New 'namespace' property - allows Namespace restriction
+          namespaces:
+            include: ["team-a"]
+          commands:
+            verbs: ["api-resources", "...", "auth"]
+            resources: ["deployments", "..", "nodes"]
+    ```
+    </td>
+    </tr>
+    </table>
+
+
+2. Communications
+    <table>
+    <tr>
+    <td> Before </td> <td> After </td>
+    </tr>
+    <tr>
+    <td>
+
+    ```yaml
+    communications:
+
+      slack:
+        token: 'SLACK_API_TOKEN'
+        # ...trimmed...
+
+
+        channel: 'SLACK_CHANNEL'
+
+
+
+
+
+
+
+
+      elasticsearch:
+        server: 'ELASTICSEARCH_ADDRESS'
+        # ...trimmed...
+
+        # ELS index settings
+        index:
+          name: botkube
+          type: botkube-event
+          shards: 1
+          replicas: 0
+
+
+
+
+
+
+    ```
+
+    </td>
+    <td>
+
+    ```yaml
+    communications: # it' is a list so it allows to have e.g multiple Elasticsearch configurations
+      - name: tenant-b-workspace
+        slack:
+          token: 'SLACK_API_TOKEN'
+          # ...trimmed...
+
+          # describe channel bindings
+          channels:
+            - name: "#team-a"
+              bindings:
+                sources:
+                  - "nodes-errors"
+                  - "deprecated-api"
+                executors:
+                  - "kubectl-read-only"
+                  - "helm-full-access"
+        elasticsearch:
+          server: 'ELASTICSEARCH_ADDRESS'
+          # ...trimmed...
+
+          # ELS index bindings
+          index:
+            - destination:
+                name: network-errors
+                type: botkube-event
+                shards: 1
+              bindings:
+                sources:
+                  - "nodes-errors"
+                  - "depreacted-api"
+                # executors - not allowed in this case, ES is "sink" only.
+    ```
+    </td>
+    </tr>
+    </table>
+
+## Use cases
+
+This section describes example configurations that enable the requested use-cases.
+
+### Route BotKube notifications to individual channels
+
+With presented configuration:
+- nodes errors are sent to a `#nodes` channel,
+- while networks errors are sent to the `#network` channel.
+
+**Communicators**
+
+```yaml
+communications: # allows to have multiple slacks, or ES configurations
+  - name: tenant-workspace
+    slack:
+      token: 'SLACK_API_TOKEN'
+      # customized notifications
+      channels:
+        - name: "#nodes"
+          bindings:
+            sources:
+              - "nodes-errors"
+        - name: "#network"
+          bindings:
+            sources:
+              - "network-errors"
+```
+
+**Sources**
+
+```yaml
+Sources:
+ - name: nodes-errors # name used for bindings
+   kubernetes:
+     resources:
+       - name: v1/nodes
+         namespaces:
+           include:
+             - @all
+         events:
+           - error
+  - name: network-errors
+    kubernetes:
```

Comment:
Should we allow referring to another source in a source to avoid duplicated configuration? Eg here we can have "all-errors" -And we can embed `network-errors`, `nodes-errors` along with other resources instead of redeclaring all again